crates/ruff_linter/src/rules/flake8_simplify/rules/ast_bool_op.rs

```diff
@@ -429,17 +442,14 @@ pub(crate) fn duplicate_isinstance_call(checker: &mut Checker, expr: &Expr) {
                     let call = node2.into();
 
                     // Generate the combined `BoolOp`.
+                    let before = values.iter().take(indices[0]).map(Clone::clone);
+                    let after = values
+                        .iter()
+                        .skip(indices[indices.len() - 1] + 1)
```

Comment:
An alternative that would avoid some index math:

```rust
let [first, .., last] = indices.as_slice() else {
    unreachable!("Indices should have at least two elements")
};
let before = values.iter().take(*first).cloned();
let after = values.iter().skip(last + 1).cloned();
```

But don't feel strongly.

Reply:
Thanks, that does feel cleaner.